packages/web/src/common/store/upload/sagas.ts

```diff
@@ -509,8 +509,10 @@ export function* handleUploads({
       } uploaded`,
       { trackIndex, stemIndex }
     )
-    const trackId = metadata.track_id
-    const stemCount = pendingStemCount[metadata.track_id]
+
+    // We know trackId exists because we generate it in uploadMultipleTracks
+    const trackId = metadata.trackId!
```

Comment:
Wouldn't hurt to explicitly check and throw though so we aren't absolutely flabbergasted if we start seeing an error about it due to a bug/regression.